source/libs/opl.h

```diff
@@ -2536,21 +2576,31 @@ void opl_midi_noteon(opl_t* opl, int channel, int note, int velocity) {
   }
 }
 
+void opl_midi_noteon(opl_t* opl, int channel, int note, int velocity) {
+  opl_midi_noteon_op2(opl, channel, note, velocity, 0);
+  opl_midi_noteon_op2(opl, channel, note, velocity, 1);
```

Comment:
There is something strange happening when playing  the doom music (option 2 in the bundled sound.c sample). Sounds like some percussion sounds go missing. I haven't looked into it in detail yet, but I suspect it might have to do with voice stealing - an interesting thing is that swapping these two lines seems to fix the issue, just by issuing the second voice first
```suggestion
  opl_midi_noteon_op2(opl, channel, note, velocity, 1);
  opl_midi_noteon_op2(opl, channel, note, velocity, 0);
```

Reply:
That makes sense to me. It would play the 2nd voice first, but those have a naturally lower priority, so if there's a lack of channels it would then immediately be overwritten by the main voice. I'll try and add some extra code so that 2nd instrumental voices don't replace primary ones too I think.